nautobot/core/tests/runner.py

```diff
@@ -30,3 +40,87 @@ def __init__(self, **kwargs):
             kwargs["exclude_tags"] = incoming_exclude_tags
 
         super().__init__(**kwargs)
+
+    def generate_report(self, test_results, result):
+        """
+        Generate Performance Report consists of unittests that are significantly slower than baseline.
+        """
+        test_result_count = len(test_results)
+
+        # Add `--generate_report` to the end of `invoke` commands to generate a report.json file consist of the performance tests result
+        if self.report_path:
+            data = {
+                "tests": [{"name": func_name, "execution_time": float(timing)} for func_name, timing in test_results],
+                "nb_tests": result.testsRun,
+                "nb_failed": len(result.errors + result.failures),
+                "total_execution_time": result.timeTaken,
+            }
+            with open(self.report_path, "w") as outfile:
+                json.dump(data, outfile)
+        # Print the results in the CLI.
+        else:
+            if test_result_count:
+                print(f"\n{test_result_count} abnormally slower tests:")
+                for func_name, timing in test_results:
+                    time = float(timing)
+                    baseline = float(self.baselines[func_name])
+                    print(f"{time:.4f}s {func_name} is significantly slower than the baseline {baseline:.4f}s")
+                # Make the test fail if there are tests significantly slower
+                assert test_result_count == 0, "Performance Tests failed due to significantly slower tests"
```

Comment:
This assertion will always fail since it's inside the `if test_result_count` block - intentional?

Reply:
Yes it is intentional. This will make the performance tests fail if any tests failed to meet the criteria derived from its baseline.